app/assets/javascripts/discourse/app/mixins/uppy-upload.js

```diff
@@ -119,6 +119,13 @@ export default Mixin.create(UppyS3Multipart, {
           this._reset();
           return false;
         }
+
+        // for a single file, we want to override file meta with the
+        // data property (which may be computed), to override any keys
+        // specified by this.data (such as name)
+        if (this.data && Object.keys(this.data).length && fileCount === 1) {
```

Comment:
Now I'm wondering: is `Object.keys(this.data).length` necessary? Imo even `this.data` part isn't strictly required (`deepMerge()` does `value || {}`)

Reply:
Ah I didn't realise `deepMerge` already did it's own fallback, will change and see

Reply:
@CvX yeah that works even with this.data set to null